File: chroma-league/src/main/java/net/booone/chromaleague/hud/parts/resource/YasuoWindBar.java
package net.booone.chromaleague.hud.parts.resource;

import net.booone.chromaleague.hud.colors.BackgroundBreathingColor;
import net.booone.chromaleague.hud.parts.ProgressBar;
import net.booone.chromaleague.state.GameStateHelper;
import net.booone.razersdk.animation.AnimatedFrame;
import net.booone.razersdk.animation.Frame;
import net.booone.razersdk.color.BreathingColor;
import net.booone.razersdk.color.Color;
import net.booone.razersdk.color.StaticColor;

public class YasuoWindBar extends AnimatedFrame {
    public static final BreathingColor WIND_SHIELD_READY_COLOR = new BackgroundBreathingColor(StaticColor.WHITE);
    public static final StaticColor WIND_SHIELD_COLOR = StaticColor.GRAY;

    @Override
    public Frame getFrame() {
        final int resourcePercentage = GameStateHelper.getResourcePercentage();
        Color color;
        if (resourcePercentage == 100) {
            color = WIND_SHIELD_READY_COLOR.getColor();
        } else {
            color = WIND_SHIELD_COLOR;
        }
        addAnimationFrame(new ProgressBar(ResourceBars.getResourceBarKeys(), resourcePercentage, color));
        return super.getFrame();
    }
}

File: src/hud/parts/resource/YasuoWindBar.ts
import {AnimatedFrame} from '../../../razer-sdk/animation/AnimatedFrame.js';
import {Frame} from '../../../razer-sdk/animation/Frame.js';
import {Color} from '../../../razer-sdk/color/Color.js';
import {StaticColor} from '../../../razer-sdk/color/StaticColor.js';
import {BackgroundBreathingColor} from '../../colors/BackgroundBreathingColor.js';
import {ProgressBar} from '../ProgressBar.js';
import {getResourceBarKeys} from './ResourceBars.js';
import {getResourcePercentage} from '../../../state/GameStateHelper.js';

export const WIND_SHIELD_READY_COLOR = new BackgroundBreathingColor(StaticColor.WHITE);
export const WIND_SHIELD_COLOR = StaticColor.GRAY;

export class YasuoWindBar extends AnimatedFrame {
    pollFrame(): Frame {
        const resourcePercentage = getResourcePercentage();
        let color: Color;
        if (resourcePercentage === 100) {
            color = WIND_SHIELD_READY_COLOR;
        } else {
            color = WIND_SHIELD_COLOR;
        }
        this.addAnimationFrame(new ProgressBar(getResourceBarKeys(), resourcePercentage, color));
        return super.pollFrame();
    }
}
